bql/planner/planner.go

```diff
@@ -639,28 +642,120 @@ func (p *queryPlan) filterOnExistence(ctx context.Context, cls *semantic.GraphCl
 	return grp.Wait()
 }
 
+// organizeClausesByBinding takes the graph clauses received as input and organize them in a map
+// on which the keys are the bindings of these clauses.
+func organizeClausesByBinding(clauses []*semantic.GraphClause) map[string][]*semantic.GraphClause {
+	clausesByBinding := map[string][]*semantic.GraphClause{}
+	for _, cls := range clauses {
+		for b := range cls.BindingsMap() {
+			clausesByBinding[b] = append(clausesByBinding[b], cls)
+		}
+	}
+
+	return clausesByBinding
+}
+
+// compatibleBindingsInClauseForFilterOperation returns a function that, for each given clause, returns the bindings that are
+// compatible with the specified filter operation.
+func compatibleBindingsInClauseForFilterOperation(operation filter.Operation) (compatibleBindingsInClause func(cls *semantic.GraphClause) (bindingsByField map[filter.Field]map[string]bool), err error) {
```

Comment:
Here "compatibleBindingsInClause" is a named return value.

Named return values are useful when your function returns two or three parameters of the same type. But using them just to avoid declaring a variable inside the function results in unnecessary API verbosity at the cost of minor implementation brevity.

Also "naked returns" are usually only acceptable only in small functions.

I'd advise instead to create a type alias for the function and also return the error as a separated parameter

Reply:
Thank you for the comment Jose. The intention here was just to document this `compatibleBindingsInClauseForFilterOperation` as best as possible, since it is a sensitive and important part of the `FILTER` implementation, using even the signature of the function for documentation as it is oriented in the link you sent above:

> "These names should be used to document the meaning of the return values."

I even started writing a new type alias for the `compatibleBindingsInClause` function but I felt that it was a bit confusing to have a separate type declared just for it, used only here, and having it named `compatibleBindingsInClauseType` or something in this direction. Also, I feel that it may be desirable for the API of `compatibleBindingsInClauseForFilterOperation` in this portion of the code to be clear and complete on what it is returning, with the function type specified there, for clarity and simplicity too.

This way, I am inclined to let the named return values here given the use-case explained above, but I agree with you regarding the naked returns. I thought before that it was a kind of guideline to use naked returns when using named return values, but I am seeing now that they do not necessarily need to come together. Then, I have just updated the function to explicitly return the values, which is also good for clarity as this function will be growing with time (as we add support for new FILTER functions on it).

If you have any other comments in this portion of the code just lmk through chat and I will be glad to make the respective changes here. Thank you once again for the comments Jose. =]

Reply:
It's clearer now, LG